### PR TITLE
Add Vector template class

### PR DESCRIPTION
**Pull request by DanRStevens**
This is to support issue #306.

### CODE REVIEW

File: proj/vs2019/NAS2D.vcxproj [hunk 1]
```diff
@@ -249,6 +249,7 @@
     <ClInclude Include="..\..\include\NAS2D\Renderer\RendererNull.h" />
     <ClInclude Include="..\..\include\NAS2D\Renderer\Color.h" />
     <ClInclude Include="..\..\include\NAS2D\Renderer\Point.h" />
+    <ClInclude Include="..\..\include\NAS2D\Renderer\Vector.h" />
```
Comment (cugone) [inline proj/vs2019/NAS2D.vcxproj]: This shouldn't be in the Renderer folder. It's a math concept and would make more sense in a `Math` subfolder.

File: proj/vs2019/NAS2D.vcxproj.filters [hunk 2]
```diff
@@ -236,6 +236,9 @@
     <ClInclude Include="..\..\include\NAS2D\Renderer\Point.h">
       <Filter>Header Files\Renderer</Filter>
     </ClInclude>
+    <ClInclude Include="..\..\include\NAS2D\Renderer\Vector.h">
+      <Filter>Header Files\Renderer</Filter>
```
Comment (cugone) [inline proj/vs2019/NAS2D.vcxproj.filters]: Ditto

File: test/test.vcxproj [hunk 3]
```diff
@@ -48,6 +48,7 @@
   </PropertyGroup>
   <ItemGroup>
     <ClCompile Include="Renderer/Rectangle.test.cpp" />
+    <ClCompile Include="Renderer/Vector.test.cpp" />
```
Comment (cugone) [inline test/test.vcxproj]: Ditto

File: include/NAS2D/Renderer/Vector.h [hunk 4]
```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+namespace NAS2D {
+
+
+template <typename BaseType>
+struct Vector {
+	BaseType x = 0;
+	BaseType y = 0;
+
```
Comment (cugone) [inline include/NAS2D/Renderer/Vector.h]: No support for 3D or 4D vectors? How is matrix math going to be handled?

File: proj/vs2019/NAS2D.vcxproj (hunk 1 above)
Comment (DanRStevens) [reply proj/vs2019/NAS2D.vcxproj]: Agreed, though I think we need to address that for all geometries at the same time, in a separate PR.

File: include/NAS2D/Renderer/Vector.h (hunk 4 above)
Comment (DanRStevens) [reply include/NAS2D/Renderer/Vector.h]: That's not a concern at this time. The class was modelled to match existing `Point` and `Rectangle` classes. We could do this at some point, but it likely needs to involve all classes at the same time. We should defer until after the other classes have been switched over to proper structs.

File: proj/vs2019/NAS2D.vcxproj (hunk 1 above)
Comment (cugone) [reply proj/vs2019/NAS2D.vcxproj]: Some PR has to make the Math folder, why not start with this one? It'll make adding more later much easier.

Comment (DanRStevens) [reply proj/vs2019/NAS2D.vcxproj]: There are inter-dependencies between the headers. Either relative includes need to be updated, which is messy, or they all move at once.

The `Vector` class is strongly related to the `Point` class and the `Rectangle` class. It should go where those are.

I would really like to move this, I just don't think this is the PR to do it in.

Comment (cugone) [reply proj/vs2019/NAS2D.vcxproj]: 'K. :)

Comment (DanRStevens) [reply proj/vs2019/NAS2D.vcxproj]: Opened issue #358 to track.

**Pull request merged**